platforms/ide/problems-api/src/test/groovy/org/gradle/api/problems/internal/ProblemIdAndProblemGroupTest.groovy

```diff
@@ -0,0 +1,65 @@
+
+    def 'name and displayName cannot be null or empty or only whitespace'() {
+        when:
+        def rootGroup = ProblemGroup.create(rootGroupName, rootGroupDisplayName)
+        def subGroup = ProblemGroup.create(subGroupName, subGroupDisplayName, rootGroup)
+        def id = ProblemId.create(idName, idDisplayName, subGroup)
+
+        then:
+        def e = thrown(IllegalArgumentException)
+        e.message == expectedMessage
+
+        where:
+        rootGroupName | subGroupName | idName  | rootGroupDisplayName | subGroupDisplayName | idDisplayName || expectedMessage
+        ""            | "valid"      | "valid" | "valid"              | "valid"             | "valid"       || 'Name must be non-blank'
+        "  "          | "valid"      | "valid" | "valid"              | "valid"             | "valid"       || 'Name must be non-blank'
+        "valid"       | ""           | "valid" | "valid"              | "valid"             | "valid"       || 'Name must be non-blank'
+        "valid"       | "  "         | "valid" | "valid"              | "valid"             | "valid"       || 'Name must be non-blank'
+        "valid"       | "valid"      | ""      | "valid"              | "valid"             | "valid"       || 'Name must be non-blank'
+        "valid"       | "valid"      | "  "    | "valid"              | "valid"             | "valid"       || 'Name must be non-blank'
+        "valid"       | "valid"      | "valid" | ""                   | "valid"             | "valid"       || 'displayName must be non-blank'
+        "valid"       | "valid"      | "valid" | "  "                 | "valid"             | "valid"       || 'displayName must be non-blank'
+        "valid"       | "valid"      | "valid" | "valid"              | ""                  | "valid"       || 'displayName must be non-blank'
+        "valid"       | "valid"      | "valid" | "valid"              | "  "                | "valid"       || 'displayName must be non-blank'
+        "valid"       | "valid"      | "valid" | "valid"              | "valid"             | ""            || 'displayName must be non-blank'
+        "valid"       | "valid"      | "valid" | "valid"              | "valid"             | "  "          || 'displayName must be non-blank'
+    }
+
+    def 'multiline names are flattened to a single line'() {
+        when:
+        def rootGroup = ProblemGroup.create("ro\no\r\nt\r\n\r\nGroupName", "ro\no\r\nt\r\n\r\nGroupDisplayName")
+        def subGroup = ProblemGroup.create("su\nb\r\nG\r\n\r\nroupName", "su\nb\r\nG\r\n\r\nroupDisplayName", rootGroup)
+        def id = ProblemId.create("id\nN\r\na\r\n\r\nme", "i\nd\r\nD\r\n\r\nisplayName", subGroup)
+
+        then:
+        id.name == 'idName'
+        id.displayName == 'idDisplayName'
+        id.group.name == 'subGroupName'
+        id.group.displayName == 'subGroupDisplayName'
+        id.group.parent.name == 'rootGroupName'
+        id.group.parent.displayName == 'rootGroupDisplayName'
```

Comment:
👏 Nice test! I think this sanitization will pay its dividends as the API will be used in more external use-cases.